src/Resolvers/Collections.ts

```diff
@@ -16,27 +16,14 @@ export class CollectionsResolver {
   // TODO: should return a connection
   @Query(returns => [CollectionCategory])
   async categories(): Promise<CollectionCategory[]> {
-    const data = await this.repository.group(
-      { category: 1 },
-      { category: { $exists: true } },
-      {},
-      (curr, result) => {
-        if (!result.collections) {
-          result.collections = [curr]
-        } else {
-          result.collections.push(curr)
-        }
-      },
-      result => result,
-      true
-    )
-
-    return data.map(({ category, collections }) => ({
+    const categories = await this.repository.distinct("category", {
+      category: { $ne: null },
+    })
+    return categories.map(category => ({
       name: category,
-      collections: collections.map(col => ({
-        ...col,
-        id: col._id,
-      })),
+      collections: this.repository
+        .find({ category })
+        .then(cols => cols.map(col => ({ ...col }))),
```

Comment:
I'm not sure I understand why that map is required here. Can you explain?

Reply:
I think you're right. At first I thought we'd have to transform the data to do things like the `id: col._id` above. But from what I can tell, that's not necessary. Pushing...

Reply:
Got it, yup that makes sense 👍